projects/packages/connection/src/class-rest-connector.php

```diff
@@ -721,14 +721,18 @@ public function connection_register( $request ) {
 	public function connection_authorize_url( $request ) {
 		$redirect_uri = $request->get_param( 'redirect_uri' ) ? admin_url( $request->get_param( 'redirect_uri' ) ) : null;
 
-		if ( ! $request->get_param( 'no_iframe' ) ) {
-			add_filter( 'jetpack_use_iframe_authorization_flow', '__return_true' );
-		}
+		if ( class_exists( 'Jetpack' ) ) {
+			$authorize_url = \Jetpack::build_authorize_url( $redirect_uri, ! $request->get_param( 'no_iframe' ) );
```

Comment:
I'm not a big fan of using the Jetpack class from within the connection package, even though I realize it would require much more work to have a clean working solution to our problem here. Maybe we should instead move the calypso env implementation to the package?